src/controllers/assistant.js

```diff
@@ -13,34 +13,137 @@
+  const { auth, log } = context;
 
   function createErrorResponse(error) {
     return createResponse({}, error.status, {
       [HEADER_ERROR]: error.message,
     });
   }
 
+  function isBase64UrlImage(base64String) {
+    return base64String.startsWith('data:image/') && base64String.endsWith('=') && base64String.includes('base64');
+  }
+
+  function validateRequestData(data) {
+    const {
+      command, prompt, imageUrl,
+    } = data;
+
+    // Validate 'command'
+    if (!command) {
+      throw new ErrorWithStatusCode('Invalid request: command is required.', STATUS.BAD_REQUEST);
+    }
+    const currentCommandConfig = commandConfig[command];
+    if (!currentCommandConfig) {
+      throw new ErrorWithStatusCode(`Invalid request: command not implemented: ${command}`, STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('prompt') && !prompt) {
+      throw new ErrorWithStatusCode('Invalid request: prompt is required.', STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('imageUrl')) {
+      if (!imageUrl) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is required.', STATUS.BAD_REQUEST);
+      }
+      // Only base64 images for now.
+      if (!isBase64UrlImage(imageUrl)) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is not a base64 encoded image.', STATUS.BAD_REQUEST);
+      }
+    }
+  }
+
+  function parseRequestContext(requestContext) {
+    if (!requestContext || !isObject(requestContext)) {
+      throw new ErrorWithStatusCode('Invalid request: missing request context.', STATUS.BAD_REQUEST);
+    }
+    if (!requestContext.data || !requestContext.attributes) {
+      throw new ErrorWithStatusCode('Invalid request: invalid request context format.', STATUS.BAD_REQUEST);
+    }
+
+    let assistantConfiguration;
+    if (!context.env.ASSISTANT_CONFIGURATION) {
+      throw new ErrorWithStatusCode('The Assistant Configuration is not defined.', STATUS.SYS_ERROR);
+    }
+    try {
+      assistantConfiguration = JSON.parse(context.env.ASSISTANT_CONFIGURATION);
+    } catch (error) {
+      throw new ErrorWithStatusCode(`Could not parse the Assistant Configuration: ${error.message}`, STATUS.SYS_ERROR);
+    }
+
+    const { data: { command, prompt, options }, attributes } = requestContext;
+    const { imageUrl, ...otherOptions } = options || {};
+    const { authInfo: { profile } } = attributes;
+    const requestData = {
+      env: {
+        ...context.env,
+        ...assistantConfiguration,
+        FIREFALL_API_KEY: assistantConfiguration.IMS_CLIENT_ID,
+      },
+      command,
+      prompt,
+      imageUrl,
+      llmModel: commandConfig[command]?.llmModel,
+      responseFormat: commandConfig[command]?.responseFormat,
+      importApiKey: requestContext.pathInfo.headers['x-api-key'],
+      apiKeyName: profile?.getName(),
+      imsOrgId: profile?.getImsOrgId() ?? requestContext.pathInfo.headers['x-gw-ims-org-id'] ?? 'N/A',
```

Comment:
Is `imsOrgId` still needed here?

Reply:
I think so.
`profile` is populated from the Importer Apikey by Spacecat (scoped apikey processor).  Checking `x-gw-ims-org-id` and then setting a default are simple backups (N/A to help debug).
I considered `imsOrgId` was part of the processed request data.
Do you think it should be somewhere else?

Reply:
I thought the gpt-client pulled the `imsOrgId` from `env` so I just wanted to make sure our Firefall requests are actually using the user's IMS Org and not the default IMS Org (aka our client id).